flatland_server/include/flatland_server/message_system/message_server.h

```diff
@@ -0,0 +1,50 @@
+#ifndef FLATLAND_SERVER_MESSAGE_SERVER_H
+#define FLATLAND_SERVER_MESSAGE_SERVER_H
+
+#include <map>
+#include <vector>
+#include <queue>
+
+namespace flatland_server {
+
+    class MessageServer;
+    template <class T>
+    class Subscriber;
+    template <class T>
+    class Publisher;
+    class MessageTopicBase {};
+
+    enum MessageType {
+        UIAction,
+        HumanAction
+    };
+
+    template <class T>
+    class MessageTopic : public MessageTopicBase {
+    public:
```

Comment:
data members of classes are supposed to be private (or protected in exceptional cases).